pkg/asset/agent/installconfig.go

```diff
@@ -169,6 +175,70 @@ func (a *OptionalInstallConfig) validateSNOConfiguration(installConfig *types.In
 	return allErrs
 }
 
+// VCenterCredentialsAreProvided returns true if server, username, password, and at least one datacenter
+// have been provided.
+func VCenterCredentialsAreProvided(vcenter vsphere.VCenter) bool {
+	if vcenter.Server != "" || vcenter.Username != "" || vcenter.Password != "" || len(vcenter.Datacenters) > 0 {
+		return true
+	}
+	return false
+}
+
+func (a *OptionalInstallConfig) validateVSpherePlatform(installConfig *types.InstallConfig) field.ErrorList {
+	var allErrs field.ErrorList
+	vspherePlatform := installConfig.Platform.VSphere
+	vcenterServers := map[string]bool{}
+	userProvidedCredentials := false
+	for _, vcenter := range vspherePlatform.VCenters {
+		vcenterServers[vcenter.Server] = true
+
+		// If any one of the required credential values is entered, then the user is choosing to enter credentials
+		if VCenterCredentialsAreProvided(vcenter) {
+			// Then check all required credential values are filled
+			userProvidedCredentials = true
+			if !(vcenter.Server != "" && vcenter.Username != "" && vcenter.Password != "" && len(vcenter.Datacenters) > 0) {
```

Comment:
Seems that this check could be simplified, e.g.
if vcenter.Server == "" || vcenter.Username == "" || vcenter.Password == "" || len(vcenter.Datacenters) == 0
or better it yet, just remove it entirely since the same checks are made just below

Reply:
Thanks Bob. I like the simplification.